campaign-launcher/client/src/components/ConnectWallet/index.tsx

```diff
@@ -8,44 +8,58 @@ import coinbaseSvg from '@/assets/coinbase.svg';
 import metaMaskSvg from '@/assets/metamask.svg';
 import walletConnectSvg from '@/assets/walletconnect.svg';
 import { useIsMobile } from '@/hooks/useBreakpoints';
+import useRetrieveSigner from '@/hooks/useRetrieveSigner';
 import { useActiveAccount } from '@/providers/ActiveAccountProvider';
-
-type Props = {
-  closeDrawer?: () => void;
-};
+import { useWeb3Auth } from '@/providers/Web3AuthProvider';
 
 const WALLET_ICONS: Record<string, string> = {
   metaMask: metaMaskSvg,
   coinbaseWalletSDK: coinbaseSvg,
   walletConnect: walletConnectSvg,
 };
 
-const ConnectWallet: FC<Props> = ({ closeDrawer }) => {
+const ConnectWallet: FC = () => {
   const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
 
   const { connectAsync, connectors } = useConnect();
   const { isConnecting } = useActiveAccount();
   const { disconnectAsync } = useDisconnect();
   const isMobile = useIsMobile();
+  const { signer } = useRetrieveSigner();
+  const { signIn, isLoading, isAuthorizing, setIsAuthorizing } = useWeb3Auth();
+
+  const isDisabled = isLoading || isConnecting;
 
   const handleConnect = async (connector: Connector) => {
     try {
+      setIsAuthorizing(true);
       await connectAsync({ connector });
     } catch (e) {
       const err = e as { message?: string; code?: number | string };
       if (err.message?.includes('Connector already connected')) {
         await disconnectAsync();
         await handleConnect(connector);
       }
+      setIsAuthorizing(false);
```

Comment:
This probably should be in `finally`?

Reply:
no, here i set it to false if something went wrong. otherwise it's getting false after the whole flow is done